### DISCUSSION TITLE
How to tell vscode-yaml to use specific schema with a yaml document?

### DISCUSSION THREAD

**Discussion by ssbarnea** [redhat-developer/vscode-yaml#431] [Q&A] [answered]
I am the maintainer of the [vscode-ansible](https://marketplace.visualstudio.com/items?itemName=zbr.vscode-ansible) extension, which depends on vscode-yaml and aims to help it correctly identify different Ansible related file types (there are at least 4-5 different schemas involved).

One major issue is that the the glob patterns provided by vscode are quite limited and in order to determine if a specific YAML should use one Ansible schema or another is a more complex process which required a bit of heuristics.

I do need some help on how can programately read and change the JSON schema of a specific YAML file from within the ansible extension, mainly helping YAML extension to use the correct schema. I do also hope that the picked schema is persistent.

Please note that altering YAML file and injecting the schema comment in file, is not a viable alternative, for multiple reasons:
* unscalable: Ansible repositories do often have hundreds if not thousands of files
* schemas are versioned and nobody expect to hardcode it inside each file
* Ansible main schemas are dynamic, being dependent on each Ansible installation environment, likely that we will endup generating a local schema for each repo and cache it instead of downloading one.

I hope that this approach would make the maintenance and development easier as users will be less likely to raise bug against YAML extension about problems that are of [broken-schema nature](https://github.com/redhat-developer/vscode-yaml/issues/423), or something else unrelated to YAML extension.

**Comment by evidolob** [comment, marked as answer]
@ssbarnea You can start looking from [schema-extension-api](https://github.com/redhat-developer/vscode-yaml/blob/master/src/schema-extension-api.ts#L6)
Thous API allows to contribute and change JSON schema on the fly.
